subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java

```diff
@@ -123,11 +122,12 @@ private void configureReportsConventionMapping(Checkstyle task, final String bas
     }
 
     private void configureToolchains(Checkstyle task) {
-        Provider<JavaLauncher> javaLauncherProvider = getToolchainService().launcherFor(new CurrentJvmToolchainSpec(project.getObjects()));
-        task.getJavaLauncher().convention(javaLauncherProvider);
         project.getPluginManager().withPlugin("java-base", p -> {
             JavaToolchainSpec toolchain = getJavaPluginExtension().getToolchain();
-            task.getJavaLauncher().convention(getToolchainService().launcherFor(toolchain).orElse(javaLauncherProvider));
+            Provider<JavaLauncher> javaLauncherProvider = getToolchainService().launcherFor(toolchain);
+            if (javaLauncherProvider.isPresent()) {
```

Comment:
I think this isn't necessary.  It's OK if the convention here overwrites the other one

Reply:
Effectively that works as before :)

Reply:
I reverted that logic to previous one